requirements.txt

```diff
@@ -1,5 +1,4 @@
+-c constraints.txt
```

Comment:
I am unclear as to whether rcbops is opinionated about the pattern of rolling the constraints file into the requirements file. This is clearly a practice published in the PIP documentation: https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format . However, the rcp-gating repo uses the pattern of treating them as discrete options: https://github.com/rcbops/rpc-gating/blob/master/rpc_jobs/build_gating_venv.yml#L61-L79

@hughsaunders or @mattt416 can tell us if there are opinions and/or guidelines for this pattern?

Reply:
I don't have any opinions on this one @johnduarte .  @odyssey4me, are you aware of any reason this shouldn't be done?

Reply:
Using constraints to specify versions, instead of requirements to do so means that all constraints can be managed in a single file, and requirements do not.

It's really a matter of convenience for managing global versions, especially when there are multiple places where requirements are set.

For example, we install pip/setuptools/wheel as requirements, using the constraints. They have to be installed independently from the other requirements because when combined together, it often fails (setuptools is most often used by the install process of other packages). We then install another set of requirements afterwards from a requirements file using the same constraints. By implementing the constraints file consistently we know that at no time will any new install change the previously installed packages. It keeps the environment consistent and predictable.

Reply:
Now that I've read the patch, I see where this is coming from. I've never seen this implementation before and have no idea whether it achieves the same goal.